docusaurus/docs/dev-docs/plugins/content-manager-apis.md

```diff
@@ -0,0 +1,295 @@
+    (panels) => [SuperImportantPanel, ...panels]
+  )
+  ```
+
+### Components
+
+You need to pass components to the API in order to add things to the Content Manager. These components are basically functions that receive some properties and return and object with some shape (depending on the function). Each component’s return object is different based on the function you’re using, but they receive similar properties, depending on whether you use a ListView or EditView API. These properties include important information about the document(s) you are viewing or editing.
+
+#### ListViewContext
+
+```jsx
+interface ListViewContext {
+  /**
+   * Will be either 'single-types' | 'collection-types'
+   */
+  collectionType: string;
+  /**
+   * The current selected documents in the table
+   */
+  documents: Document[];
+  /**
+   * The current content-type's model.
+   */
+  model: string;
+}
+```
+
+#### EditViewContext
+
+```jsx
+interface EditViewContext {
+  /**
+   * This will only be null if the content-type
+   * does not have draft & publish enabled.
+   */
+  activeTab: 'draft' | 'published' | null;
+  /**
+   * Will be either 'single-types' | 'collection-types'
+   */
+  collectionType: string;
+  /**
+   * Will be undefined if someone is creating an entry.
+   */
+  document?: Document;
+  /**
+   * Will be undefined if someone is creating an entry.
+   */
+  documentId?: string;
+  /**
+   * Will be undefined if someone is creating an entry.
+   */
+  meta?: DocumentMetadata;
+  /**
+   * The current content-type's model.
+   */
+  model: string;
+}
+```
+
+:::tip
+More information about types and APIs can be found in [Strapi's codebase, in the `/admin/src/content-manager.ts` file](https://github.com/strapi/strapi/blob/v5/main/packages/core/content-manager/admin/src/content-manager.ts).
+:::
+
+**Example:**
+
+Adding a panel to the sidebar can be done this way:
+
+```jsx title="my-plugin/components/my-panel.ts"
+import type { PanelComponent, PanelComponentProps } from '@strapi/content-manager/strapi-admin';
+
+const Panel: PanelComponent = ({ 
+  activeTab, 
+  collectionType, 
+  document, 
+  documentId, 
+  meta, 
+  model 
+}: PanelComponentProps) => {
+  return {
+    title: 'My Panel',
+    content: <p>I'm on {activeTab}</p>
+  }
+}
+```
+
+## Available APIs
+
+### `addEditViewSidePanel`
+
+Use this to add new panels to the Edit view sidebar, just like in the following example where something is added to the Releases panel:
+
+![addEditViewSidePanel](/img/assets/content-manager-apis/add-edit-view-side-panel.png)
+
+```jsx
+addEditViewSidePanel(panels: DescriptionReducer<PanelComponent> | PanelComponent[])
+```
+
+#### PanelDescription
+
+The interface of the API only receives the following 2 properties:
+
+```jsx
+{
+  title: string;
+  content: React.ReactNode;	
+}
+```
+
+### `addDocumentAction`
+
+Use this API to add more actions to the Edit view of the Content Manager. There are 3 positions available:
```

Comment:
With this API you can also add options on the ListView (if one of the positions is table-row)

Reply:
Thanks, I didn't understand indeed what table-row was referring to!

Reply:
Fixed ✅